test/integration/without_iam/gremlin/test_gremlin_metadata.py

```diff
@@ -0,0 +1,59 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+"""
+
+import time
+import pytest
+
+from test.integration import DataDrivenGremlinTest
+from graph_notebook.magics.metadata import build_gremlin_metadata_from_query
+
+
+class TestMetadataClassFunctions(DataDrivenGremlinTest):
+
+    @pytest.mark.gremlin
+    def test_gremlin_default_query_metadata(self):
+        query = "g.V().has('airport','code','CZM').out('route').path().by('code')"
+        query_start = time.time() * 1000
+        results = self.client.gremlin_query(query)
+        query_time = time.time() * 1000 - query_start
+        gremlin_metadata = build_gremlin_metadata_from_query(query_type='query', results=results, query_time=query_time)
+        meta_dict = gremlin_metadata.to_dict()
+
+        self.assertEqual(meta_dict["Query mode"], "query")
+        self.assertIsInstance(meta_dict["Request execution time (ms)"], float)
+        self.assertEqual(meta_dict["# of results"], 11)
+        self.assertIsInstance(meta_dict["Response size (bytes)"], int)
+
+    @pytest.mark.gremlin
```

Comment:
Let's add `@pytest.mark.neptune` for any explain/profile tests since those are neptune-specific

Reply:
Good catch, added.